### PR TITLE
[WIP] Migration to actix_web 1.0.7

### PR DESCRIPTION
**Pull request by SergeiMal**
## Overview

First iteration. without testkit

---

See: https://jira.bf.local/browse/ECR-XYZW

### Definition of Done

- [ ] There are no TODOs left in the merged code
- [ ] Change is covered by automated tests
- [ ] Benchmark results are attached (if applicable)
- [ ] The [coding guidelines] are followed
- [ ] Public API has proper documentation
- [ ] Changelog is updated if needed (in case of notable or breaking changes)
- [ ] The continuous integration build passes

[coding guidelines]: https://github.com/exonum/exonum/blob/master/CONTRIBUTING.md#conventions

### CODE REVIEW

File: exonum/src/api/backends/actix.rs [hunk 1]
```diff
@@ -101,21 +98,32 @@ impl ApiBackend for ApiBuilder {
...
+                actix_web::http::Method::GET => output.service(
+                    actix_web::web::resource(&handler.name)
+                        .route(actix_web::web::get().to(move |r, p| inner(r, p))),
+                ),
+                actix_web::http::Method::POST => output.service(
+                    actix_web::web::resource(&handler.name)
+                        .route(actix_web::web::post().to(move |r, p| inner(r, p))),
+                ),
+                _ => {
+                    // TODO: log error. unrichable ?
```
Comment (alekseysidorov) [inline exonum/src/api/backends/actix.rs]: It seems that raw web handlers can have abritrary methods.

File: exonum/src/runtime/api.rs [hunk 2]
```diff
@@ -97,18 +97,19 @@ impl ServiceApiScope {
...
-                    InstanceDescriptor {
-                        id: descriptor.0,
-                        name: descriptor.1.as_ref(),
-                    },
-                );
-                let result = handler(&state, query);
-                Box::new(result.into_future())
-            });
+
+        let endp_handl = move |query: Q| -> crate::api::FutureResult<I> {
```
Comment (popzxc) [inline exonum/src/runtime/api.rs]: Do we have to economize symbols? :)

File: exonum/src/api/manager.rs [hunk 3]
```diff
@@ -86,10 +85,10 @@ impl ApiManager {
 
     fn initiate_restart(&mut self, manager: Addr<Self>) {
         info!("Restarting servers.");
-        for (addr, config) in self.api_runtime_addresses.drain() {
+        for (addr, config) in self.api_runtime_addresses.drain(..) {
             let manager = manager.clone();
             Arbiter::spawn(
-                addr.send(StopServer { graceful: true })
+                addr.stop(true)
```
Comment (popzxc) [inline exonum/src/api/manager.rs]: I'd move `true` to a variable named `graceful` to have a more readable code.

**Comment by popzxc** [comment]
This could be closed I think?

**Pull request closed without merging**